File: go/374.go
func guessNumber(n int) int {

    left := 1
    right := n

    for true {
        mid := (left + right) / 2

        result := guess(mid)

        switch result {
        case -1:
            right = mid - 1
        case 0:
            return mid
        case 1:
            left = mid + 1
        }
    }
    return -1
}

File: py/374.py
# Author: Andrew Hamara

# Solution for leetcode problem 374. Guess number higher or lower

class Solution:
    def guessNumber(self, n:int) -> int:
        low = 1
        high = n
        while True:
            mid = int((low + high)/2)
            if guess(mid) == 1:
                low = mid + 1
            elif guess(mid) == -1:
                high = mid - 1
            else:
                return mid
